db/db_impl/db_impl_files.cc

```diff
@@ -797,11 +833,10 @@ Status DBImpl::FinishBestEffortsRecovery() {
   assert(versions_->GetColumnFamilySet());
   ColumnFamilyData* default_cfd = versions_->GetColumnFamilySet()->GetDefault();
   assert(default_cfd);
-  // Even if new_descriptor_log is false, we will still switch to a new
-  // MANIFEST and update CURRENT file, since this is in recovery.
+  // Switch to a new MANIFEST.
   s = versions_->LogAndApply(
       default_cfd, *default_cfd->GetLatestMutableCFOptions(), &edit, &mutex_,
-      directories_.GetDbDir(), /*new_descriptor_log*/ false);
+      directories_.GetDbDir(), /*new_descriptor_log*/ true);
```

Comment:
By setting `true` to `false` here (keep original code unchanged), we can save one extra MANIFEST/CURRENT switch.

Reply:
In `write_dbid_to_manifest = true` case.

Reply:
I didn't notice that `VersionSet::descriptor_log_` is opened lazily. So when writing db id, even if `new_descriptor_log` is `false`, it will switch to a new manifest. If not writing db id, the same logic applies here, even if it's false, a new manifest will be created.
Thanks for catching this.